compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1588,11 +1588,19 @@ private BType computeIntersectionType(BLangIntersectionTypeNode intersectionType
 
         BLangType bLangTypeOne = constituentTypeNodes.get(0);
         BType typeOne = resolveTypeNode(bLangTypeOne, env);
+
+        if (typeOne == symTable.noType) {
```

Comment:
When can these types be `noType`?

Reply:
If the member type definition is after intersection type definition, this can happen. It is handled later on via unresolved types.

```ballerina
type MyType Foo & readonly;

type Foo record {
    int i;
};
```